### ISSUE TITLE
Suggest to add checkbox for mass selection

### ISSUE THREAD

**Issue by paul-cylix** [AndreSouzaAbreu/vue-data-table#53]
I really like this page vue data table. i hope that you can add a feature like this. it means a lot to us. who wants mass selection.
thank you and stay safe.

**Comment by Asinging** [comment]
The reason i love this plugin is the fact it has many flexible tweaks you can implement. you can write a custom component of checkbox and append it to the table. then you have an array collecting the data from the  checked event of the of the checkbox

**Comment by paul-cylix** [comment]
i will try this one sir @Asinging. thank you for the advice

**Comment by AndreSouzaAbreu** [comment]
> The reason i love this plugin is the fact it has many flexible tweaks you can
> implement. you can write a custom component of checkbox and append it to the
> table. then you have an array collecting the data from the checked event of
> the of the checkbox

This is actually possible indeed. I will add an example on README of how to use
set up custom components for mass-selection using checkboxes.

**Comment by AndreSouzaAbreu** [comment]
Here is the demo I made very quickly:

https://codesandbox.io/s/vue-data-table-demo04-qyetb7

**Comment by paul-cylix** [comment]
sir thank you this helps me a lot. i want to ask if there is a way to literally select all the rows per page by single pressing a button.
scenario:
I am in the table with 5 pagination and I go to page 2 and I add a button select all but when I press that it will select all the rows that only seen on that specific page this will also help the user of my system. is it possible sir?. thank you and stay safe

**Comment by AndreSouzaAbreu** [comment]
> sir thank you this helps me a lot. i want to ask if there is a way to
> literally select all the rows per page by single pressing a button. scenario:
> I am in the table with 5 pagination and I go to page 2 and I add a button
> select all but when I press that it will select all the rows that only seen
> on that specific page this will also help the user of my system. is it
> possible sir?. thank you and stay safe

It is possible to select all visible entries in the table, and it is indeed
very easy. Just add a button outside of the table and add a `@onclick` vue
directive to call some function (i.e, `selectVisibleEntries()`) to select the
visible entries.

You could use a quirk hack here: the custom `footerComponent` has access to all
visible entries, so you could loop through the entries of the current
pagination inside the `footerComponent`, and for each entry set `selected` to
`true`, which will toggle the checkboxes.

It seems to me this should be enough.

**Comment by paul-cylix** [comment]
Yes, sir sorry, I didn't realize that I raise this question to you and you have already solved it. thank you for always helping me
